web/src/message_edit_history.ts

```diff
@@ -164,56 +186,86 @@ export function fetch_and_render_message_history(message: Message): void {
                 }
 
                 content_edit_history.push(item);
-                prev_time = time;
             }
             if (prev_stream_item !== null) {
                 assert(message.type === "stream");
                 prev_stream_item.new_stream = sub_store.maybe_get_stream_name(message.stream_id);
             }
-            $("#message-history").attr("data-message-id", message.id);
-            $("#message-history").html(
-                render_message_edit_history({
-                    edited_messages: content_edit_history,
-                }),
-            );
+
+            // in order to update the header bar colors for message row inside the overlay
+            // we have to reverse the array, lets say [edited, edited in stream1, moved to stream2 ,edited in stream2]
+            // is our content_edit_history.
+            if (message.is_stream) {
+                // get stream color for current stream i.e last stream message was moved in
+                let stream_color: string = get_color(message.stream_id);
+                let recipient_bar_color: string = get_recipient_bar_color(stream_color);
+                for (const message_object of content_edit_history.toReversed()) {
```

Comment:
Call this variable `edit_history_entry`, not `message_object`, to better hint its type.